scripts/commands/addfish.lua

```diff
@@ -0,0 +1,88 @@
+-----------------------------------
+-- func: addfish <fishId> <length> <weight>
+-- desc: Adds a measured fish to the GMs inventory.
+-----------------------------------
+require('scripts/globals/fishing_contest')
+-----------------------------------
+local commandObj = {}
+
+commandObj.cmdprops =
+{
+    permission = 3,
+    parameters = 'sii'
+}
+
+local function error(player, msg)
+    player:printToPlayer(msg)
+    player:printToPlayer('!addfish <fish> <length> <weight> <optional:ranked>')
+end
+
+commandObj.onTrigger = function(player, item, length, weight, ranked)
+    -- Load needed text ids for players current zone..
+    local ID = zones[player:getZoneID()]
+    local itemToGet = 0
+    ranked = ranked or 0
+
+    -- validate item
+    if item == nil then
+        -- No Item Provided
+        error(player, 'No Item ID given.')
+        return
+    elseif item ~= nil and tonumber(item) == nil then
+        -- Item was provided, but was not a number.  Try text lookup.
+        local retItems = utils.filterArray(xi.fishingcontest.fish, function(_, fish)
+            return fish.name == item
+        end)
+
+        if #retItems ~= 1 then
+            player:printToPlayer(string.format('Item %s not found in fish table.', item))
+            return
+        end
+
+        itemToGet = retItems[1].id
+    else
+        -- Number was provided, so just use it
+        itemToGet = tonumber(item)
+    end
+
+    -- At this point, if there's no item found, exit out of the function
+    if itemToGet == 0 then
+        error(player, 'Item not found.')
+        return
+    end
+
+    -- Confirm the size/weight
+    if
+        length == 0 or
+        length == nil or
+        weight == 0 or
+        weight == nil
+    then
+        error(player, 'Size/Weight not provided.')
+        return
+    end
+
+    -- TODO: check qty and stack size + remaining inventory space instead of hardcoded == 0 check
+    -- Ensure the GM has room to obtain the item...
+    if player:getFreeSlotsCount() == 0 then
+        player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, itemToGet)
+        return
+    end
+
+    -- Give the GM the item...
+    local obtained = player:addItem({ id = itemToGet,
+                                        quantity = 1,
+                                        exdata =
+                                        {
+                                            [0] = bit.band(length, 0x00FF),
+                                            [1] = bit.rshift(bit.band(length, 0xFF00), 8),
+                                            [2] = bit.band(weight, 0x00FF),
+                                            [3] = bit.rshift(bit.band(weight, 0xFF00), 8),
+                                            [4] = ranked,
+                                        } })
+    if obtained then
+        player:messageSpecial(ID.text.ITEM_OBTAINED, itemToGet)
+    end
```

Comment:
Can/should this be done with npcUtils?

Reply:
I would like it to be but I'd need to add code to npc_util to accept exdata.  I can certainly do that if you want me to build it into this PR.